File: Demo/src/main/java/com/bloom/demo/service/patient/PatientJoinService.java
package com.bloom.demo.service.patient;

import com.bloom.demo.model.employee.Doctor;
import com.bloom.demo.model.patient.Patient;
import com.bloom.demo.model.patient.PatientRoomDetails;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface PatientJoinService {

    Mono<PatientRoomDetails> getPatientRoomDetails(String patientId);

    Mono<Doctor> getPatientDoctor(String patientId);

    Flux<Patient> getDoctorPatients(String doctorId);

}

File: Front-End/src/app/patient-join.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {PatientRoomDetails} from './model/PatientRoomDetails';
import {Doctor} from './model/Doctor';

@Injectable({
  providedIn: 'root'
})
export class PatientJoinService {

  private patientJoinURL = 'http://localhost:8080/patient/';

  constructor(private http: HttpClient) {
  }

  getPatientRoomDetails(patientId: number): Observable<PatientRoomDetails> {
    return this.http.get<PatientRoomDetails>(this.patientJoinURL + 'room/' + patientId);
  }

  getPatientDoctor(patientId: number): Observable<Doctor> {
    return this.http.get<Doctor>(this.patientJoinURL + 'doctor/' + patientId);
  }
}
